packages/xforms-engine/src/client/RankNode.ts

```diff
@@ -0,0 +1,50 @@
+import type { RankType, RankControlDefinition } from '../parse/body/control/RankControlDefinition.ts';
+import type { LeafNodeDefinition } from '../parse/model/LeafNodeDefinition.ts';
+import type { BaseValueNode, BaseValueNodeState } from './BaseValueNode.ts';
+import type { RootNode } from './RootNode.ts';
+import type { TextRange } from './TextRange.ts';
+import type { GeneralParentNode } from './hierarchy.ts';
+import type { LeafNodeValidationState } from './validation.ts';
+
+export interface RankItem {
+	get label(): TextRange<'item-label'>;
+	get value(): string;
+}
+
+export type RankValueOptions = readonly RankItem[];
+
+export interface RankNodeState extends BaseValueNodeState<readonly string[]> {
+	get valueOptions(): RankValueOptions;
+
+	/**
+	 * An ordered collection of values from {@link RankItem}.
+	 * The order of the items is important and must be preserved during processing.
+	 */
+	get value(): readonly string[];
+}
+
+export interface RankDefinition extends LeafNodeDefinition<V> {
```

Comment:
```suggestion
export interface RankDefinition<V extends ValueType = ValueType> extends LeafNodeDefinition<V> {
```

This probably got jumbled up in the mixup around select value types, but `RankDefinition` should have a `ValueType` type parameter, like `SelectDefinition` does in `SelectNode`. This is somewhat important so the types match up with the runtime check (in `RankControl`) to enforce only supporting the `'string'` value type. (I'll leave a corresponding suggestion there.)

Reply:
Fixed!